### PR TITLE
Expose prior_box op into detection.py

### PR DESCRIPTION
**Pull request by baiyfbupt**
Fix https://github.com/PaddlePaddle/Paddle/issues/10774

### CODE REVIEW

**Comment by qingqing01** [comment]
Need to add documentation.

File: python/paddle/fluid/layers/detection.py [hunk 1]
```diff
@@ -564,6 +565,48 @@ def __reshape_to_2d(var):
...
+def prior_box(input,
+              image,
+              min_sizes,
+              max_sizes,
+              aspect_ratios,
+              variance,
+              flip=False,
+              clip=False,
+              step_w=0.0,
+              step_h=0.0,
```
Comment (qingqing01) [inline python/paddle/fluid/layers/detection.py]: how about merge `step_w`, `step_h` into one `steps`?

File: python/paddle/fluid/layers/detection.py [hunk 2]
```diff
@@ -564,6 +565,48 @@ def __reshape_to_2d(var):
     return loss
 
 
+def prior_box(input,
+              image,
+              min_sizes,
+              max_sizes,
+              aspect_ratios,
```
Comment (qingqing01) [inline python/paddle/fluid/layers/detection.py]: Sometimes, `aspect_ratios` is not necessary, the user may not specify this argument.

```
aspect_ratios=None
```

File: python/paddle/fluid/layers/detection.py [hunk 3]
```diff
@@ -564,6 +565,48 @@ def __reshape_to_2d(var):
...
+              variance,
```
Comment (qingqing01) [inline python/paddle/fluid/layers/detection.py]: give default value for `variance `.

File: python/paddle/fluid/layers/detection.py (hunk 1 above)
Comment (baiyfbupt) [reply python/paddle/fluid/layers/detection.py]: good idea, it will be done

**Comment by qingqing01** [comment]
Need to add unit test in https://github.com/PaddlePaddle/Paddle/blob/develop/python/paddle/fluid/tests/test_detection.py

File: python/paddle/fluid/layers/detection.py (hunk 1 above)
Comment (baiyfbupt) [reply python/paddle/fluid/layers/detection.py]: Done

File: python/paddle/fluid/layers/detection.py [hunk 4]
```diff
@@ -564,6 +565,48 @@ def __reshape_to_2d(var):
     return loss
 
 
+def prior_box(input,
+              image,
+              min_sizes,
+              max_sizes,
```
[the remark this answers is not in the document]
Comment (baiyfbupt) [reply python/paddle/fluid/layers/detection.py]: Done

File: python/paddle/fluid/layers/detection.py (hunk 2 above)
Comment (baiyfbupt) [reply python/paddle/fluid/layers/detection.py]: Done

File: python/paddle/fluid/layers/detection.py (hunk 3 above)
Comment (baiyfbupt) [reply python/paddle/fluid/layers/detection.py]: Done

**Comment by baiyfbupt** [comment]
@qingqing01 Done

**Pull request merged**